x-pack/plugins/gis/public/elasticsearch_geo_utils.js

```diff
@@ -217,12 +202,11 @@ export function createExtentFilter(mapExtent, geoFieldName, geoFieldType) {
 }
 
 /*
- * Convert map bounds to envelope
- * Bounds that cross the dateline are split into 2 envelopes
+ * Convert map bounds to polygon
  */
-export function convertMapExtentToEnvelope({ maxLat, maxLon, minLat, minLon }) {
+export function convertMapExtentToPolygon({ maxLat, maxLon, minLat, minLon }) {
```

Comment:
I wouldn't write this as a recursive function, since the recursion is guaranteed to stop at the 2nd call. 

I'd just extract the base-case at the end in it's own utility-function that is not exported. e.g. something like `formatEnvelopeAsPolygon` or something.